docs/content/tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,249 @@
+
+# adding the word cloud asset
+@asset
+def topstories_word_cloud(topstories):
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN", "S"])
+    titles_text = " ".join([str(item) for item in topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Since we're saving the asset, this converts the word cloud into an image
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    return image_data # returns the image instead of showing it with `plt.show()`
+```
+
+---
+
+## Step 3: Educating users with metadata
+
+Software-defined assets can be enriched with different types of metadata. Anything can be used as metadata for an SDA. Common details to add are:
+
+- Statistics about the data, such as row counts or other data profiling
+- Test results or assertions about the data
+- Images or tabular previews of the asset
+- Information about who owns the asset, where it's stored, and links to external documentation
+
+The following code adds a row count and a preview of the `topstories` asset. It uses the <PyObject module="dagster" object="Output" displayText="Output"/> class from Dagster to bundle the asset and its metadata together. Update your code for the `topstories` asset to match the changes below.
+
+```python
+from dagster import asset, get_dagster_logger, Output, MetadataValue
+# add an import to the Output and MetadataValue classes from the dagster library
+
+@asset
+def topstories(topstory_ids):
+    logger = get_dagster_logger()
+
+    results = []
+    for item_id in topstory_ids:
+        item = requests.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json"
+        ).json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            logger.info(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return Output(  # The return value is updated to wrap it in `Output` class
+        value=df,   # The original df is passed in with the `value` parameter
+        metadata={
+            "num_records": len(df), # Metadata can be any key-value pair
+            "preview": MetadataValue.md(df.head().to_markdown()),
+            # The `MetadataValue` class has useful static methods to build Metadata
+        }
+    )
+```
+
+Reload the definitions and re-materialize your assets. The metadata can then be seen in the following places:
+
+- In the **Asset graph** page, click on an asset and its metadata will be shown in the right sidebar:
+
+    <Image
+        alt="the asset's materialization metadata"
+        src="/images/tutorial/asset-metadata.png"
+        width={1152}
+        height={666}
+    />
```

Comment:
```suggestion
    <Image
    alt="the asset's materialization metadata"
    src="/images/tutorial/asset-metadata.png"
    width={1152}
    height={666}
    />
```

I think this is what's breaking in one of the BK tests. Pretty sure all lines need to be indented at the same level or it throws a fit.